### PR TITLE
#837 decompose name search inputs

### LINKED ISSUE #837: Decompose `NameSearchModal` / `NameSearchInput`
- There are two components: `NameSearchModal` and `NameSearchInput` in the `Name` module.
- It would be good to decompose these components to reduce the eventual complexity and number of props:
    - `CustomerSearch..`
    - `SupplierSearch..`
    - `InternalSupplierSearch..`
    - ... other name variants (i.e. in the future `PatientNameSearch..`

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #837 

- I think this is a good idea
- Adds `Customer`, `Supplier` and `InternalSupplier` Search inputs and modals in the hope it will reduce conditional code in the future and lead to a preference for specific 'typed' components i.e. patient, manufacturer. I'm pretty confident that these will become different enough to warrant the change 😁 
- Used these components in the places they should be - i.e. internal supplier for request requisition

### CODE REVIEW

File: packages/system/src/Name/utils.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+export const simpleNameOptionRenderer: AutocompleteOptionRenderer<
+  NameRowFragment
+> = (props, item) => (
+  <DefaultAutocompleteItemOption {...props}>
+    <Typography sx={{ marginInlineEnd: '10px', fontWeight: 'bold', width: 75 }}>
+      {item.code}
+    </Typography>
+    <Typography>{item.name}</Typography>
+  </DefaultAutocompleteItemOption>
+);
```
Comment (joshxg) [inline packages/system/src/Name/utils.tsx:31]: Is this a good idea? 😁  thought it looked ok..

File: packages/system/src/Name/Components/CustomerSearchModal/CustomerSearchModal.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  const { data, isLoading } = useCustomers();
+  const t = useTranslation(['app', 'common']);
+
+  return (
+    <ListSearch
+      loading={isLoading}
+      open={open}
+      options={data?.nodes ?? []}
+      onClose={onClose}
+      title={t('suppliers')}
```
Comment (mark-prins) [inline packages/system/src/Name/Components/CustomerSearchModal/CustomerSearchModal.tsx:20]:
```suggestion
      title={t('customers')}
```

File: packages/system/src/Name/Components/SupplierSearchModal/SupplierSearchModal.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import { useSuppliers, NameRowFragment } from '../../api';
+import { NameSearchProps } from '../../utils';
+
+export const SupplierSearchModal: FC<NameSearchProps> = ({
+  open,
+  onClose,
+  onChange,
+}) => {
+  const { data, isLoading } = useSuppliers();
+  const t = useTranslation(['app', 'common']);
```
Comment (mark-prins) [inline packages/system/src/Name/Components/SupplierSearchModal/SupplierSearchModal.tsx:12]:
```suggestion
  const t = useTranslation('app');
```

File: packages/system/src/Name/Components/CustomerSearchModal/CustomerSearchModal.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import { useCustomers, NameRowFragment } from '../../api';
+import { NameSearchProps } from '../../utils';
+
+export const CustomerSearchModal: FC<NameSearchProps> = ({
+  open,
+  onClose,
+  onChange,
+}) => {
+  const { data, isLoading } = useCustomers();
+  const t = useTranslation(['app', 'common']);
```
Comment (mark-prins) [inline packages/system/src/Name/Components/CustomerSearchModal/CustomerSearchModal.tsx:12]:
```suggestion
  const t = useTranslation('app');
```

File: packages/system/src/Name/Components/InternalSupplierSearchModal/InternalSupplierSearchModal.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import { useInternalSuppliers, NameRowFragment } from '../../api';
+import { NameSearchProps } from '../../utils';
+
+export const InternalSupplierSearchModal: FC<NameSearchProps> = ({
+  open,
+  onClose,
+  onChange,
+}) => {
+  const { data, isLoading } = useInternalSuppliers();
+  const t = useTranslation(['app', 'common']);
```
Comment (mark-prins) [inline packages/system/src/Name/Components/InternalSupplierSearchModal/InternalSupplierSearchModal.tsx:12]:
```suggestion
  const t = useTranslation('app');
```

File: packages/system/src/Name/utils.tsx (hunk 1 above)
Comment (mark-prins) [reply packages/system/src/Name/utils.tsx:31]: great idea 😁

**Pull request merged**